tests/e2e/vsphere_volume_expansion.go

```diff
@@ -199,8 +238,440 @@ var _ = ginkgo.Describe("Volume Expansion Test", func() {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(), fmt.Sprintf("Volume %q is not detached from the node %q", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+		}()
+
+		increaseVolumeMultipleTimes(f, client, namespace, volHandle, pvclaim, pod)
+	})
+
 })
 
+//increaseVolumeMultipleTimes this method increases the same volume multiple times and verifies PVC and Filesystem size
+func increaseVolumeMultipleTimes(f *framework.Framework, client clientset.Interface, namespace string, volHandle string, pvclaim *v1.PersistentVolumeClaim, pod *v1.Pod) {
+	// Modify PVC spec to trigger volume expansion
+	ginkgo.By("Expanding pvc 10 times")
+	currentPvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+	newSize := currentPvcSize.DeepCopy()
+	for i := 0; i < 10; i++ {
+		newSize.Add(resource.MustParse("1Gi"))
+		ginkgo.By(fmt.Sprintf("Expanding pvc to new size: %v", newSize))
+		pvclaim, err := expandPVCSize(pvclaim, newSize, client)
+		framework.ExpectNoError(err, "While updating pvc for more size")
+		gomega.Expect(pvclaim).NotTo(gomega.BeNil())
+
+		pvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+		if pvcSize.Cmp(newSize) != 0 {
+			framework.Failf("error updating pvc %q to size %v", pvclaim.Name, newSize)
+		}
+	}
+
+	ginkgo.By("Waiting for controller resize to finish")
+	framework.Logf("PVC name : " + pvclaim.Name)
+	pv := getPvFromClaim(client, namespace, pvclaim.Name)
+	pvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+	waitForPvResize(pv, client, pvcSize, totalResizeWaitPeriod)
+
+	ginkgo.By("Checking for conditions on pvc")
+	framework.Logf("PVC Name :", pvclaim.Name)
+	pvclaim, err := waitForPVCToReachFileSystemResizePendingCondition(client, namespace, pvclaim.Name, totalResizeWaitPeriod)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By(fmt.Sprintf("Invoking QueryCNSVolumeWithResult with VolumeID: %s", volHandle))
+	queryResult, err := e2eVSphere.queryCNSVolumeWithResult(volHandle)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	if len(queryResult.Volumes) == 0 {
+		err = fmt.Errorf("QueryCNSVolumeWithResult returned no volume")
+	}
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	ginkgo.By("Verifying disk size requested in volume expansion is honored")
+	newSizeInMb := int64(12288)
+	if queryResult.Volumes[0].BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails).CapacityInMb != newSizeInMb {
+		err = fmt.Errorf("Received wrong disk size after volume expansion. Expected: %d Actual: %d", newSizeInMb, queryResult.Volumes[0].BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails).CapacityInMb)
+	}
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	framework.ExpectNoError(waitForPVCToCompleteResizing(client, namespace, pvclaim.Name, poll, totalResizeWaitPeriod))
+	ExpectedSizeInMb := int64(12288)
+	framework.ExpectNoError(verifyPVCSizeAfterResize(f, client, namespace, pvclaim.Name, ExpectedSizeInMb))
```

Comment:
this can be replaced with `waitForFSResize`, I don't see a need for `waitForPVCToCompleteResizing` and `verifyPVCSizeAfterResize` methods